packages/apify/src/actor.ts

```diff
@@ -609,13 +617,29 @@ export class Actor<Data extends Dictionary = Dictionary> {
      *
      * @param item Object or array of objects containing data to be stored in the default dataset.
      * The objects must be serializable to JSON and the JSON representation of each object must be smaller than 9MB.
+     * @param eventName If provided, the method will attempt to charge for the event for each pushed item.
      * @ignore
      */
-    async pushData(item: Data | Data[]): Promise<void> {
+    async pushData(item: Data | Data[]): Promise<void>;
+    async pushData(item: Data | Data[], eventName: string): Promise<ChargeResult>;
+    async pushData(item: Data | Data[], eventName?: string | undefined): Promise<ChargeResult | void> {
         this._ensureActorInit('pushData');
 
         const dataset = await this.openDataset();
-        return dataset.pushData(item);
+
+        const maxChargedCount = eventName !== undefined ? this.chargingManager.calculateMaxEventChargeCountWithinLimit(eventName) : Infinity;
+        const toCharge = Array.isArray(item) ? item.length : 1;
+
+        if (toCharge > maxChargedCount) {
+            const items = Array.isArray(item) ? item : [item];
+            await dataset.pushData(items.slice(0, toCharge));
+        } else {
+            await dataset.pushData(item);
+        }
+
+        if (eventName) {
+            return await this.chargingManager.charge({ eventName, count: Math.min(toCharge, maxChargedCount) });
```

Comment:
```suggestion
            return this.chargingManager.charge({ eventName, count: Math.min(toCharge, maxChargedCount) });
```

Reply:
I mean, sure. Does this make any practical difference, other than making the line shorter?

Reply:
Cool, thanks for the explanation!